02/distributed/lib/distributed.ex

```diff
@@ -0,0 +1,50 @@
+defmodule Distributed do
+  @moduledoc """
+  Documentation for `Distributed`.
+  """
+
+  alias Distributed.Router
+
+  @doc """
+  Run calc fib function on remote node
+
+  ## Examples
+
+      iex> Distributed.calc_fib(5)
+      {15, :"n2@127.0.0.1"}
+
+  """
+
+  @spec calc_fib(non_neg_integer()) :: {non_neg_integer(), atom()}
+  def calc_fib(num \\ 5) do
+    next = GenServer.call(Router, :take_node)
+
+    next_node =
+      cond do
+        next == node() -> GenServer.call(Router, :take_node)
+        true -> next
+      end
```

Comment:
`cond/1` обычно не нужен примерно никогда, он гораздо медленнее паттерн-матчинга и неуклюжий, используйте только как последний вариант.

```suggestion
      case node() do
        ^next -> GenServer.call(Router, :take_node)
        _ -> next
      end
```